### DISCUSSION TITLE
Allow remix in nx monorepo to work without package.json

### DISCUSSION THREAD

**Discussion by itxch** [remix-run/remix#6821] [Proposals]
Hi all,

  I've recently began a project where I am using remix with an `nx` monorepo. `nx` has a experimental remix plugin: https://nx.dev/recipes/other/remix, [code here](https://github.com/nrwl/nx-labs/tree/main/packages/remix).

But one of the issues is that remix **needs** a `package.json` file to work, while `nx` proposes to have 1 global package.json - and this discussion isn't really to weigh up the pros/cons of each approach, its more to allow users to have the flexibility to go with either.

The code that blocks this from working, from what I can tell is part of the `remix-dev` package:
- https://github.com/remix-run/remix/blob/main/packages/remix-dev/config.ts#L591 
- https://github.com/remix-run/remix/blob/main/packages/remix-dev/cli/commands.ts#L323

The current implementation throws an error if a package.json doesn't exist.

A simple solution would be to allow users to specify the directory their `package.json` is in, relative to the `remix.config.js` file.

Would be good to get peoples thoughts on this before opening a PR 

**EDIT:** an alternative approach would be to use [pkg-up](https://www.npmjs.com/package/pkg-up) as a fallback where there is no package.json in the remix root, shout out to @rossipedia for the suggestion

**Comment by th-m** [comment]
I have been using remix, nx at work for the last 6 months. Allowing the user to specify the directory in `remix.config.js` would be a delightful solution for my needs.

I have hosted my remix / nx applications on netlify and vercel, and I think the proposed solution would also simplify deployments on those platforms.

**Reply by itxch** [reply]
Are you guys going with the integrated nx repo approach too?

Also its nice to see other people also using Remix and nx we're just starting to adopt it at work, would be great to see how its going for you guys, if your open to it please reach out [via twitter ](https://twitter.com/itxchh) .

**Comment by SahilMahadwar** [comment]
+1

**Comment by binaryartifex** [comment]
+1 we also use an NX integrated monorepo and being forced to use a package.json per remix app goes against the grain of why we chose an integrated monorepo in the first place. surely this could be as simple as a config setting as mentioned above?

**Comment by jvliwanag** [comment]
At least with the vite version of remix, remix seems to work fine on an integrated repo. on the vite config per project, just need to the ff vite config.

```typescript
export default defineConfig({
  root: path.resolve(__dirname, "../.."), // resolves to workspace dir
  plugins: [
    remix({
      appDirectory: path.resolve(__dirname, "app"),
    }),
    tsconfigPaths(),
  ],
});

```

And execute the equivalent of:

```
remix vite:dev -c apps/myapp/vite.config.ts
```

Don't forget to clear out `apps/myapp/node_modules` when doing this.

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
